src/Compiler/CodeGen/IlxGen.fs

```diff
@@ -11039,15 +11250,25 @@ and GenTypeDef cenv mgbuf lazyInitInfo eenv m (tycon: Tycon) =
                                 Some(g.ilg.typ_Object.TypeSpec)
 
                         let ilMethodDef =
-                            mkILSimpleStorageCtorWithParamNames (
+                            (mkILSimpleStorageCtorWithParamNames (
                                 spec,
                                 ilThisTy,
                                 [],
                                 ChooseParamNames fieldNamesAndTypes,
                                 reprAccess,
                                 None,
                                 eenv.imports
-                            )
+                            ))
+                                .With(
+                                    customAttrs =
+                                        mkILCustomAttrs
+                                            [
+                                                GetDynamicDependencyAttribute
+                                                    cenv
+                                                    0x660 (*Public and NonPublic Fields and Properties*)
```

Comment:
Wondering: Is the dynamic generation of the full enum definition needed, when all we need is to pass in a constant value of of 0x660 ?
Would the trimmer still work correctly if we simply had an int with the value of 0x660 instead?

Reply:
@T-Gro, we generate a typed member on the attribute that is typed with the enum, that is used by the trimmer.  I don't think there is much benefit, in having the type have a surface area that does not match the correct dotnet framework time that it represents.

Reply:
Yes, I understand ... I agree it would be nice, but it would confuse the heck out of someone looking at the il :-)